perses/annihilation/relative.py

```diff
@@ -2629,3 +2629,1504 @@ def _handle_exceptions(self):
+    are defined with `lambda_{i}_{electrostatics/sterics}_exceptions_{old/new}`.
+
+    NOTE : the `render_bool_string` yields lambda-independent environment region per particle parameters.
+           environment region per particle parameters should be old/new system independent.
+           Instead of interpolating between old and new parameters that are identical, we use a select statement that uses the following logic (with electrostatic charges as an example);
+            if both regions are envronment, charge1 is given by charge_old1 (the old system charge of the first particle);
+            otherwise, charge1 is charge_old1 * {old_bool_string} + charge_new1 * {new_bool_string}.
+                notably, this introduces the necessity that the lambda protocol completely turns off (to 0) the charge_old1 (via {old_bool_string}) before charging charge_new1 (via {new_bool_string}).
+    The same strategy is also used for torsions. see `self._transcribe_torsions`
+
+    NOTE : we need to add add rxn field electrostatic configuration-independent offsets (U^{off}, U^{slf}) given in eqs 5,6 in DOI: 10.1039/D0CP03835K
+    """
+    from openmmtools.constants import ONE_4PI_EPS0
+    _default_RF_expr_list = [
+                             "U_electrostatics;", # define scaled electrostatic term
+                             "U_electrostatics = switch * c_RF * u_RF;", # define product electrostatic term
+
+                             #define switching function
+                             "switch = step(-t) + step(t) * step(1-t) * (1 + (t^3) * (-10 + t * (15 - (6 * t))));",
+                             "t = (r_eff - r_switch) / (r_cutoff - r_switch);",
+
+                             #define topological term
+                             f"c_RF = {ONE_4PI_EPS0} * chargeProd;",
+                             "chargeProd = charge1 * charge2;",
+                             "charge1 = ((charge_old1 * {old_bool_string1}) + (charge_new1 * {new_bool_string1})) * {scale_bool_string1};", 
+                             "charge2 = ((charge_old2 * {old_bool_string2}) + (charge_new2 * {new_bool_string2})) * {scale_bool_string2};",
+
+                             #define reaction field u_RF
+                             "u_RF = (1 / r_eff) + ((eps_RF - 1) * (r_eff^2)) / ((r_cutoff^3) * (1 + (2 * eps_RF)));",
+
+                             #r_eff
+                             "r_eff = sqrt(r^2 + w^2);",
+
+                             #4th dimension:
+                             #"w = step(unique_new1 + unique_new2 + unique_old1 + unique_old2 - 0.1) * r_cutoff * ({old_bool_string1} - {new_bool_string1}) * {w_scale};",
+                             "w = is_unique_old * r_cutoff * w_scale * (1 - lambda_0_electrostatics_old) + is_unique_new * w_scale * (1 - lambda_0_electrostatics_new);",
```

Comment:
Why does `r_cutoff * w_scale` appear in the first term but only `w_scale` in the second term?